src/app/application.cpp

```diff
@@ -495,26 +497,42 @@ int Application::exec(const QStringList &params)
     GuiIconProvider::initInstance();
 #endif
 
-    BitTorrent::Session::initInstance();
-    connect(BitTorrent::Session::instance(), &BitTorrent::Session::torrentFinished, this, &Application::torrentFinished);
-    connect(BitTorrent::Session::instance(), &BitTorrent::Session::allTorrentsFinished, this, &Application::allTorrentsFinished, Qt::QueuedConnection);
+    try {
+        BitTorrent::Session::initInstance();
+        connect(BitTorrent::Session::instance(), &BitTorrent::Session::torrentFinished, this, &Application::torrentFinished);
+        connect(BitTorrent::Session::instance(), &BitTorrent::Session::allTorrentsFinished, this, &Application::allTorrentsFinished, Qt::QueuedConnection);
 
 #ifndef DISABLE_COUNTRIES_RESOLUTION
-    Net::GeoIPManager::initInstance();
+        Net::GeoIPManager::initInstance();
 #endif
-    ScanFoldersModel::initInstance(this);
+        ScanFoldersModel::initInstance(this);
 
 #ifndef DISABLE_WEBUI
-    m_webui = new WebUI;
+        m_webui = new WebUI;
 #ifdef DISABLE_GUI
-    if (m_webui->isErrored())
-        return 1;
-    connect(m_webui, &WebUI::fatalError, this, []() { QCoreApplication::exit(1); });
+        if (m_webui->isErrored())
+            return 1;
+        connect(m_webui, &WebUI::fatalError, this, []() { QCoreApplication::exit(1); });
 #endif // DISABLE_GUI
 #endif // DISABLE_WEBUI
 
-    new RSS::Session; // create RSS::Session singleton
-    new RSS::AutoDownloader; // create RSS::AutoDownloader singleton
+        new RSS::Session; // create RSS::Session singleton
+        new RSS::AutoDownloader; // create RSS::AutoDownloader singleton
+    }
+    catch (const RuntimeError &err) {
+#ifdef DISABLE_GUI
+        fprintf(stderr, "%s", err.what());
+#else
+        QMessageBox msgBox;
+        msgBox.setIcon(QMessageBox::Critical);
+        msgBox.setText(tr("Application failed to start."));
```

Comment:
Should this be the msgBox title instead? (.setWindowTitle())
If so, why not initialize all these things in the constructor?

Reply:
From QMessageBox documentation:
>Two APIs for using QMessageBox are provided, the property-based API, and the static functions. Calling one of the static functions is the simpler approach, but it is less flexible than using the property-based API, and the result is less informative. **Using the property-based API is recommended.**

>A message box displays a primary **text** to alert the user to a situation, an **informative text** to further explain the alert or to ask the user a question, and an optional **detailed text** to provide even more data if the user requests it.
...
This is the approach recommended in the macOS Guidelines. Similar guidelines apply for the other platforms, ...
The *Microsoft Windows User Interface Guidelines* recommend using the **application name as the window's title**

So I would like not only to keep this approach here, but also to modify all the existing code to follow the interface guidelines further.

But if you have the opposite opinion, then I will degrade it to using the "static method" API.

Reply:
can you provide a screenshot?

Reply:
@glassez No, you misunderstood me.
About my point 1: I was confused in case you wanted to have it as a title. Now I know you had a reason. I don't totally disagree with that, so I'll pass.
About my point 2: I meant the 2nd constructor. Not the static methods. But that is combined with point 1.

>So I would like not only to keep this approach here, **but also to modify all the existing code to follow the interface guidelines further.**

IMO, I wouldn't like that. But we can have this discussion in a separate issue with example screenshots.

Reply:
>I wouldn't like that. 

Then would it make sense to leave it in its current form here?
Please make a suggestion how it should be and I will change it so that this detail doesn't interfere with merging the main thing.

Reply:
@glassez go ahead with this PR as it is..

>IMO, I wouldn't like that. But we can have this discussion in a separate issue with example screenshots.

Actually I might change my stance on this. I saw some of the screenshots in the Qt docs, and it wasn't bad. If you want to pursue the change in the whole codebase, please open an issue first with a couple of screenshots showcasing the differences.

Reply:
Ok. Let's not stumble here.
Besides, it's not that important.
Although, can @thalieht wants to participate?

Reply:
I don't have an opinion on this.